internal/pkg/graph/graph.go

```diff
@@ -401,19 +369,21 @@ func (t *graphTraversal[V]) execute(ctx context.Context, lg *LabeledGraph[V]) er
 		for _, vertex := range vertices {
 			vertex := vertex
 			// Delay processing this vertex if any of its dependent vertices are yet to be processed.
-			if len(t.filterPreviousVerticesByStatus(graph, vertex, t.nextVertexSkipStatus)) != 0 {
+			if len(t.filterPreviousVerticesByStatus(graph, vertex, unvisited)) != 0 {
```

Comment:
interesting - we start processing the vertex even if some of its depending vertices are still being "visiting". Why do we not wait for all of them to be "visited?